From ed97e900a7775cd14202807e20ad796547dee9f5 Mon Sep 17 00:00:00 2001
From: Alon Rotman <alon.rotman@solid-run.com>
Date: Thu, 29 Oct 2020 17:25:22 +0200
Subject: [PATCH] u-boot: disable board_fix_fdt on cn913x com epress type 7

Signed-off-by: Alon Rotman <alon.rotman@solid-run.com>

---
 board/Marvell/octeontx2_cn913x/board.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/board/Marvell/octeontx2_cn913x/board.c b/board/Marvell/octeontx2_cn913x/board.c
index 457f9d4a36..e0998faf28 100644
--- a/board/Marvell/octeontx2_cn913x/board.c
+++ b/board/Marvell/octeontx2_cn913x/board.c
@@ -67,6 +67,12 @@ int board_init(void)
 	return 0;
 }
 
+int board_fix_fdt (void *fdt)
+{
+        return 0;
+}
+
+
 #if (CONFIG_IS_ENABLED(OCTEONTX_SERIAL_BOOTCMD) ||	\
 	CONFIG_IS_ENABLED(OCTEONTX_SERIAL_PCIE_CONSOLE)) &&	\
 	!CONFIG_IS_ENABLED(CONSOLE_MUX)
