From 3a695f5a42abb6bee509a760dba59bebcb4ff7e8 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <sergey.temerkhanov@auriga.com>
Date: Mon, 23 Mar 2015 18:22:40 +0300
Subject: [PATCH 0312/1239] Add network commands

---
 include/configs/thunderx_88xx.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/configs/thunderx_88xx.h b/include/configs/thunderx_88xx.h
index f178081eb5..689219751a 100644
--- a/include/configs/thunderx_88xx.h
+++ b/include/configs/thunderx_88xx.h
@@ -90,6 +90,14 @@
 
 #define CONFIG_CMD_I2C
 
+#define CONFIG_CMD_DHCP
+#define CONFIG_CMD_MII
+#define CONFIG_CMD_NET
+#define CONFIG_CMD_PING
+#define CONFIG_CMD_TFTP
+#define CONFIG_CMD_TFTPPUT
+#define CONFIG_CMD_NFS
+
 #define CONFIG_CMD_FAT
 #define CONFIG_CMD_EXT2
 #define CONFIG_CMD_EXT4
-- 
2.29.0

