From 7b7750feb3e0385f048a44c1420621a3a8b65e1e Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Wed, 4 Nov 2015 14:39:33 -0800
Subject: [PATCH 0332/1239] net: e1000: Fix PHY reset check on 8257x

---
 drivers/net/e1000.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/net/e1000.c b/drivers/net/e1000.c
index f7bf2a4151..d5818d188e 100644
--- a/drivers/net/e1000.c
+++ b/drivers/net/e1000.c
@@ -4571,6 +4571,7 @@ e1000_get_phy_cfg_done(struct e1000_hw *hw)
 			mdelay(1);
 			timeout--;
 		}
+/*
 		if (!timeout) {
 			DEBUGOUT("MNG configuration cycle has not "
 					"completed.\n");
@@ -4584,6 +4585,7 @@ e1000_get_phy_cfg_done(struct e1000_hw *hw)
 			 * return -E1000_ERR_RESET;
 			 */
 		}
+*/
 		break;
 	}
 
@@ -5147,7 +5149,7 @@ fill_rx(struct e1000_hw *hw)
 	rd = rx_base + rx_tail;
 	rx_tail = (rx_tail + 1) % 8;
 	memset(rd, 0, 16);
-	rd->buffer_addr = cpu_to_le64((unsigned long)packet);
+	rd->buffer_addr = cpu_to_le64((uintptr_t)packet);
 
 	/*
 	 * Make sure there are no stale data in WB over this area, which
@@ -5236,7 +5238,6 @@ e1000_configure_tx(struct e1000_hw *hw)
 		E1000_WRITE_REG(hw, TARC1, tarc);
 	}
 
-
 	e1000_config_collision_dist(hw);
 	/* Setup Transmit Descriptor Settings for eop descriptor */
 	hw->txd_cmd = E1000_TXD_CMD_EOP | E1000_TXD_CMD_IFCS;
@@ -5247,7 +5248,6 @@ e1000_configure_tx(struct e1000_hw *hw)
 	else
 		hw->txd_cmd |= E1000_TXD_CMD_RS;
 
-
 	if (hw->mac_type == e1000_igb) {
 		E1000_WRITE_REG(hw, TCTL_EXT, 0x42 << 10);
 
@@ -5257,11 +5257,7 @@ e1000_configure_tx(struct e1000_hw *hw)
 		mdelay(20);
 	}
 
-
-
 	E1000_WRITE_REG(hw, TCTL, tctl);
-
-
 }
 
 /**
@@ -5366,7 +5362,7 @@ _e1000_poll(struct e1000_hw *hw)
 
 	if (!(rd->status & E1000_RXD_STAT_DD))
 		return 0;
-	/* DEBUGOUT("recv: packet len=%d\n", rd->length); */
+	DEBUGOUT("recv: packet len=%d\n", rd->length);
 	/* Packet received, make sure the data are re-loaded from RAM. */
 	len = le16_to_cpu(rd->length);
 	invalidate_dcache_range((unsigned long)packet,
-- 
2.29.0

