From 801fbd87fb3ac10f87df1080669bdf55ffa8cfe0 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Wed, 4 Nov 2015 16:47:28 -0800
Subject: [PATCH] Add CSR include directory symlink

---
 csr | 1 +
 1 file changed, 1 insertion(+)
 create mode 120000 csr

diff --git a/csr b/csr
new file mode 120000
index 0000000000..5702a41e9a
--- /dev/null
+++ b/csr
@@ -0,0 +1 @@
+../../csr
\ No newline at end of file
