From 0476b2ebdd1fd2ef6f9baa74b4d549f5eb16de15 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Mon, 9 Nov 2015 14:20:14 -0800
Subject: [PATCH 0365/1239] cavium: Enable every function enumeration in config

---
 include/configs/thunderx_88xx.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/thunderx_88xx.h b/include/configs/thunderx_88xx.h
index 42b1f69d6c..164b68d2b6 100644
--- a/include/configs/thunderx_88xx.h
+++ b/include/configs/thunderx_88xx.h
@@ -196,6 +196,7 @@
 #define CONFIG_PCI_SCAN_SHOW
 #undef CONFIG_PCI_ENUM_ONLY
 #define CONFIG_PCI_PNP
+#define CONFIG_SYS_PCI_ARI
 #define CONFIG_THUNDERX_ECAMS 8
 #define CONFIG_THUNDERX_RCS 12
 
-- 
2.29.0

