From db63ada85a5aeca8540175ca022c3dca2fe503ad Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Sun, 8 May 2016 00:03:05 -0700
Subject: [PATCH 0412/1239] Fix lmac_count for XAUI/XLAUI interface

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/cavium/thunder_bgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index bcb15a6576..fd4ffc6a91 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -624,7 +624,7 @@ static void bgx_init_hw(struct bgx *bgx)
 				continue;
 			lmac->lmac_type = 1;
 			lmac->lane_to_sds = 0xE4;
-			lmac_count = 4;
+			lmac_count = 1;
 			break;
 		case QLM_MODE_RXAUI:
 			lmac->lmac_type = 2;
@@ -641,7 +641,7 @@ static void bgx_init_hw(struct bgx *bgx)
 				continue;
 			lmac->lmac_type = 4;
 			lmac->lane_to_sds = 0xE4;
-			lmac_count = 4;
+			lmac_count = 1;
 			break;
 		case QLM_MODE_10G_KR:
 			lmac->lmac_type = 3;
-- 
2.29.0

