From 87925d5ae7b522ca68f3537eec7cf4301190e56f Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Fri, 13 May 2016 08:59:16 -0700
Subject: [PATCH] Update BGX_SPUX_STATUS1.rcv_lnk only when the value is 0

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>

---
 drivers/net/cavium/thunder_bgx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 3d0afdaa60..3d919dd2ec 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -593,7 +593,8 @@ static int bgx_xaui_check_link(struct lmac *lmac)
 	}
 
 	/* Receive link is latching low. Force it high and verify it */
-	bgx_reg_modify(bgx, lmacid, BGX_SPUX_STATUS1, SPU_STATUS1_RCV_LNK);
+	if (!(bgx_reg_read(bgx, lmacid, BGX_SPUX_STATUS1 & SPU_STATUS1_RCV_LNK)))
+		bgx_reg_modify(bgx, lmacid, BGX_SPUX_STATUS1, SPU_STATUS1_RCV_LNK);
 	if (bgx_poll_reg(bgx, lmacid, BGX_SPUX_STATUS1,
 			 SPU_STATUS1_RCV_LNK, false)) {
 		printf("SPU receive link down\n");
