From de8f5ee61ef402fcbe783ff21ebc1fd521e28732 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Wed, 29 Jun 2016 22:55:01 -0700
Subject: [PATCH 0512/1239] Disabled debugging messages

---
 board/cavium/thunderx/thunderx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/cavium/thunderx/thunderx.c b/board/cavium/thunderx/thunderx.c
index b1e8080052..55d3d535c7 100644
--- a/board/cavium/thunderx/thunderx.c
+++ b/board/cavium/thunderx/thunderx.c
@@ -2,7 +2,7 @@
 /**
  * (C) Copyright 2014, Cavium Inc.
 **/
-#define DEBUG
+
 #include <common.h>
 #include <dm.h>
 #include <malloc.h>
-- 
2.29.0

