From 243a79525d2907f64788895297de12c4efc9ceca Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Mon, 12 Sep 2016 19:09:52 -0700
Subject: [PATCH 0524/1239] net:thunderx: Fix compilation errors

---
 drivers/net/cavium/thunder_bgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 181971c7b3..297b988207 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -885,7 +885,7 @@ static void bgx_init_hw(struct bgx *bgx)
 					lmac->lane_to_sds = 0xe;
 					lmac_count++;
 				}
-				continue
+				continue;
 			} else
 				continue;
 			break;
@@ -1143,7 +1143,7 @@ int thunderx_bgx_probe(struct udevice *dev)
 	bgx->bgx_id = (node * CONFIG_MAX_BGX_PER_NODE) + bgx_idx;
 
 	for (lmac = 0; lmac < MAX_LMAC_PER_BGX; lmac += 2) {
-		if (is_altpkg && (lmac == 2) && (bgx_id == 0)) {
+		if (is_altpkg && (lmac == 2) && (bgx_idx == 0)) {
 			qlm[lmac - 1] = get_qlm_for_bgx(node, bgx_idx, lmac);
 			debug("qlm[%d] = %d\n", lmac, qlm[lmac-1]);
 		} else {
-- 
2.29.0

