From eb72213921df0757ad0d9760f4b8216371337bc6 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Mon, 14 Nov 2016 03:52:26 -0800
Subject: [PATCH 0565/1239] Clarify what the CRC7 displayed for mmc info is
 for.

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
---
 drivers/mmc/cavium_mmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/cavium_mmc.c b/drivers/mmc/cavium_mmc.c
index 357cfdee0b..6bc24a7cbf 100644
--- a/drivers/mmc/cavium_mmc.c
+++ b/drivers/mmc/cavium_mmc.c
@@ -2019,7 +2019,7 @@ void print_mmc_device_info(struct mmc *mmc)
 		       (mmc->cid[3] >> 12) & 0xf,
 		       ((mmc->cid[3] >> 8) & 0xf) + start_year);
 	}
-	printf("CRC7:                  0x%02x (%svalid)\n", mmc->cid[3] & 0xff,
+	printf("CID CRC7:              0x%02x (%svalid)\n", mmc->cid[3] & 0xff,
 	       (mmc->cid[3] & 0xff) == mmc_crc7_32(mmc->cid, 4) ? "" : "in");
 	printf("Capacity:              %llu bytes (%llu blocks)\n",
 	       mmc->capacity, mmc->capacity / mmc->read_bl_len);
-- 
2.29.0

