From 634b4e94be3def53867477327b7e196450e03f25 Mon Sep 17 00:00:00 2001
From: Nithya <nithya.mani@caviumnetworks.com>
Date: Thu, 31 Aug 2017 11:08:55 -0700
Subject: [PATCH 0701/1239] t93 : update maxcpus = 24 to support multi core
 booting

Signed-off-by: Nithya <nithya.mani@caviumnetworks.com>
---
 include/configs/thunderx_93xx.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/thunderx_93xx.h b/include/configs/thunderx_93xx.h
index 6c16676cb0..09d0232c26 100644
--- a/include/configs/thunderx_93xx.h
+++ b/include/configs/thunderx_93xx.h
@@ -249,7 +249,7 @@
 #define CONFIG_BOOTARGS			\
 					"console=ttyAMA0,115200n8 " \
 					"earlycon=pl011,0x87e028000000 " \
-					"debug maxcpus=1 rootwait rw "\
+					"debug maxcpus=24 rootwait rw "\
 					"root=/dev/sda2 coherent_pool=16M"
 
 /** Store U-Boot version in "ver" environment variable */
-- 
2.29.0

