From 66e831959825ef3951b900016483a83f105b0db8 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <sergey.temerkhanov@auriga.com>
Date: Thu, 12 Mar 2015 19:10:33 +0300
Subject: [PATCH 0306/1239] Added a DDR3 module type string returning function

---
 common/ddr_spd.c  | 34 ++++++++++++++++++++++++++++++++++
 include/ddr_spd.h |  2 ++
 2 files changed, 36 insertions(+)

diff --git a/common/ddr_spd.c b/common/ddr_spd.c
index 58dc9b3781..da63a1884d 100644
--- a/common/ddr_spd.c
+++ b/common/ddr_spd.c
@@ -82,6 +82,40 @@ crc16(char *ptr, int count)
 	return crc & 0xffff;
 }
 
+const char *ddr3_modtype_str(const ddr3_spd_eeprom_t *spd)
+{
+	switch(spd->module_type) {
+	case DDR3_SPD_MODULETYPE_RDIMM:
+		return "Registered DIMM";
+	case DDR3_SPD_MODULETYPE_UDIMM:
+		return "Unbuffered DIMM";
+	case DDR3_SPD_MODULETYPE_SO_DIMM:
+		return "SODIMM";
+	case DDR3_SPD_MODULETYPE_MICRO_DIMM:
+		return "Micro DIMM";
+	case DDR3_SPD_MODULETYPE_MINI_RDIMM:
+		return "Mini RDIMM";
+	case DDR3_SPD_MODULETYPE_MINI_UDIMM:
+		return "Mini UDIMM";
+	case DDR3_SPD_MODULETYPE_MINI_CDIMM:
+		return "Mini CDIMM";
+	case DDR3_SPD_MODULETYPE_72B_SO_UDIMM:
+		return "ECC Unbuffered SODIMM";
+	case DDR3_SPD_MODULETYPE_72B_SO_RDIMM:
+		return "ECC Registered SODIMM";
+	case DDR3_SPD_MODULETYPE_72B_SO_CDIMM:
+		return "ECC CDIMM";
+	case DDR3_SPD_MODULETYPE_LRDIMM:
+		return "LRDIMM";
+	case DDR3_SPD_MODULETYPE_16B_SO_DIMM:
+		return "16B SODIMM";
+	case DDR3_SPD_MODULETYPE_32B_SO_DIMM:
+		return "32B SODIMM";
+	}
+
+	return "Unknown module";
+}
+
 unsigned int
 ddr3_spd_check(const ddr3_spd_eeprom_t *spd)
 {
diff --git a/include/ddr_spd.h b/include/ddr_spd.h
index fe163da43e..e4313ce4b5 100644
--- a/include/ddr_spd.h
+++ b/include/ddr_spd.h
@@ -504,6 +504,8 @@ extern void ddr2_spd_dump(const ddr2_spd_eeprom_t *spd);
 extern unsigned int ddr3_spd_check(const ddr3_spd_eeprom_t *spd);
 unsigned int ddr4_spd_check(const struct ddr4_spd_eeprom_s *spd);
 
+const char *ddr3_modtype_str(const ddr3_spd_eeprom_t *spd);
+
 /*
  * Byte 2 Fundamental Memory Types.
  */
-- 
2.29.0

