From 709d6d83e29598bbaa6deabc5cf7f95322f718d6 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <sergey.temerkhanov@auriga.com>
Date: Tue, 7 Apr 2015 21:49:56 +0300
Subject: [PATCH 0321/1239] Add missing definitions

---
 include/configs/thunderx_88xx.h | 55 +++++++++++++++++++++++++++++++++
 1 file changed, 55 insertions(+)

diff --git a/include/configs/thunderx_88xx.h b/include/configs/thunderx_88xx.h
index 11055dca20..033a349211 100644
--- a/include/configs/thunderx_88xx.h
+++ b/include/configs/thunderx_88xx.h
@@ -22,6 +22,61 @@
 
 #define CONFIG_SYS_LOWMEM_BASE		MEM_BASE
 
+#define CONFIG_SYS_MEM_MAP		{{0x000000000000UL, 0x40000000000UL, \
+					  PTL2_MEMTYPE(MT_NORMAL) |	     \
+					  PTL2_BLOCK_NON_SHARE},	     \
+					 {0x800000000000UL, 0x40000000000UL, \
+					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
+					  PTL2_BLOCK_NON_SHARE},	     \
+					 {0x840000000000UL, 0x40000000000UL, \
+					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
+					  PTL2_BLOCK_NON_SHARE},	     \
+					 {0x880000000000UL, 0x40000000000UL, \
+					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
+					  PTL2_BLOCK_NON_SHARE},	     \
+					  MT_DEVICE_NGNRNE}, 		     \
+					 {0xffffffffffffUL, 0xfffffffffffUL, \
+					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
+					  PTL2_BLOCK_NON_SHARE},	     \
+					}
+
+#define CONFIG_SYS_MEM_MAP_SIZE		4
+
+#define CONFIG_COREID_MASK		0xffffff
+#define CONFIG_SYS_FULL_VA
+
+#define CONFIG_SYS_VA_BITS		48
+#define CONFIG_SYS_PTL2_BITS		42
+#define CONFIG_SYS_BLOCK_SHIFT		29
+#define CONFIG_SYS_PTL1_ENTRIES		64
+#define CONFIG_SYS_PTL2_ENTRIES		8192
+
+#define CONFIG_SYS_PGTABLE_SIZE		\
+	((CONFIG_SYS_PTL1_ENTRIES + \
+	  CONFIG_SYS_MEM_MAP_SIZE * CONFIG_SYS_PTL2_ENTRIES) * 8)
+#define CONFIG_SYS_TCR_EL1_IPS_BITS	(5UL << 32)
+#define CONFIG_SYS_TCR_EL2_IPS_BITS	(5 << 16)
+#define CONFIG_SYS_TCR_EL3_IPS_BITS	(5 << 16)
+
+#define CONFIG_SYS_PCI_MEM_CPU	{0x881040000000UL, 0x885040000000UL, 0x889040000000UL,\
+				 0x891040000000UL, 0x895040000000UL, 0x899040000000UL}
+
+#define CONFIG_SYS_PCI_PREF_CPU	{0x882080000000UL, 0x886080000000UL, 0x88a080000000UL,\
+				 0x892080000000UL, 0x896080000000UL, 0x89a080000000UL}
+
+#define CONFIG_SYS_PCI_IO_CPU	{0x883000000000UL, 0x887000000000UL, 0x88b000000000UL,\
+				 0x893000000000UL, 0x897000000000UL, 0x89b000000000UL}
+
+#define CONFIG_SYS_PCI_MEM_BUS	 0x40000000
+#define CONFIG_SYS_PCI_PREF_BUS	 0x80000000
+#define CONFIG_SYS_PCI_IO_BUS	 0x0
+
+#define CONFIG_SYS_PCI_MEM_SIZE  0x40000000
+#define CONFIG_SYS_PCI_PREF_SIZE 0x80000000
+#define CONFIG_SYS_PCI_IO_SIZE   0x10000
+
+#define CONFIG_SLT
+
 /* Link Definitions */
 #define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x7fff0)
 
-- 
2.29.0

