From c4a81e5492a4efca067b598dd16df2e6e594f075 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <sergey.temerkhanov@auriga.com>
Date: Tue, 7 Apr 2015 21:50:20 +0300
Subject: [PATCH 0322/1239] Fix fdt.c build warnings

---
 board/cavium/thunderx/fdt.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/board/cavium/thunderx/fdt.c b/board/cavium/thunderx/fdt.c
index 5cee3a6e37..571d652772 100644
--- a/board/cavium/thunderx/fdt.c
+++ b/board/cavium/thunderx/fdt.c
@@ -9,14 +9,14 @@
 #include <errno.h>
 #include <linux/compiler.h>
 
-#ifdef CONFIG_OF_LIBFDT
- #include <libfdt.h>
- #include <fdt_support.h>
-#endif
+#include <libfdt.h>
+#include <fdt_support.h>
+#include <cavium/atf.h>
 
 #if defined(CONFIG_OF_LIBFDT)
 
 #ifdef CONFIG_THUNDER_BGX
+#include <cavm-csr.h>
 /**
  * To remove unwanted nodes from fdt .
  *
@@ -126,6 +126,8 @@ static void ft_setup_bgx(char *fdt, unsigned int node, unsigned int bgx_id)
 	}
 }
 
+#define VNIC_PER_NODE 2
+
 static void ft_setup_macs(void *fdt, int node)
 {
 	int nodeoffset, addr, len, ret, i;
-- 
2.29.0

