From 9e105c741005fd29dfdc7128b847617718a566cf Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Tue, 3 Nov 2015 19:10:45 -0800
Subject: [PATCH 0327/1239] cavium: Add hardware watchdog support

---
 board/cavium/thunderx/thunderx.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/board/cavium/thunderx/thunderx.c b/board/cavium/thunderx/thunderx.c
index 4e81159982..9a7543d7b7 100644
--- a/board/cavium/thunderx/thunderx.c
+++ b/board/cavium/thunderx/thunderx.c
@@ -203,6 +203,27 @@ int board_eth_init(bd_t *bis)
 
 #endif
 
+	rc = pci_eth_init(bis);
+
 	return rc;
 }
 
+#ifdef CONFIG_HW_WATCHDOG
+void hw_watchdog_reset(void)
+{
+	ssize_t node, core;
+
+	for (node = 0; node < atf_node_count(); node++)
+		for (core = 0; core < thunderx_core_count(); core++)
+			writeq(~0ULL, CSR_PA(node, GTI_CWD_POKE(core)));
+}
+
+void hw_watchdog_disable(void)
+{
+	ssize_t node, core;
+
+	for (node = 0; node < atf_node_count(); node++)
+		for (core = 0; core < thunderx_core_count(); core++)
+			writeq(0ULL, CSR_PA(node, GTI_CWD_WDOG(core)));
+}
+#endif
-- 
2.29.0

