From d291e4d7cb05055e519d7306b9867f1e23d4ac8c Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Mon, 9 Nov 2015 11:24:20 -0800
Subject: [PATCH 0364/1239] cavium: Update CPU addresses to match bus regions

---
 include/configs/thunderx_88xx.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/configs/thunderx_88xx.h b/include/configs/thunderx_88xx.h
index 6165611b57..42b1f69d6c 100644
--- a/include/configs/thunderx_88xx.h
+++ b/include/configs/thunderx_88xx.h
@@ -66,10 +66,10 @@
 #define CONFIG_SYS_TCR_EL2_IPS_BITS	(5 << 16)
 #define CONFIG_SYS_TCR_EL3_IPS_BITS	(5 << 16)
 
-#define CONFIG_SYS_PCI_MEM_CPU	{0x881040000000UL, 0x885040000000UL, 0x889040000000UL,\
-				 0x891040000000UL, 0x895040000000UL, 0x899040000000UL}
+#define CONFIG_SYS_PCI_MEM_CPU	{0x881010000000UL, 0x885010000000UL, 0x889010000000UL,\
+				 0x891010000000UL, 0x895010000000UL, 0x899010000000UL}
 
-#define CONFIG_SYS_PCI_PREF_CPU	{0x882080000000UL, 0x886080000000UL, 0x88a080000000UL,\
+#define CONFIG_SYS_PCI_PREF_CPU	{0x882100000000UL, 0x886100000000UL, 0x88a100000000UL,\
 				 0x892080000000UL, 0x896080000000UL, 0x89a080000000UL}
 
 #define CONFIG_SYS_PCI_IO_CPU	{0x883000000000UL, 0x887000000000UL, 0x88b000000000UL,\
-- 
2.29.0

