From 6af5dd5b8594ac3916d174b7b74f038c2ac7147a Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Thu, 19 Nov 2015 10:43:21 -0800
Subject: [PATCH 0371/1239] cavium: Add CONFIG_SYS_VSNPRINTF

---
 include/configs/thunderx_88xx.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/thunderx_88xx.h b/include/configs/thunderx_88xx.h
index a1447fbe3d..8974fba8a1 100644
--- a/include/configs/thunderx_88xx.h
+++ b/include/configs/thunderx_88xx.h
@@ -22,6 +22,8 @@
 
 #define CONFIG_SYS_BOOTM_LEN (16 << 20)
 
+#define CONFIG_SYS_VSNPRINTF
+
 #define CONFIG_IDENT_STRING	\
 	" for Cavium Thunder CN88XX ARM v8 Multi-Core"
 #define CONFIG_BOOTP_VCI_STRING		"Diagnostics"
-- 
2.29.0

