From ac93c127e8ad4400e793739d71745fb6c7d6a2f5 Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Fri, 29 Jan 2016 15:08:23 -0800
Subject: [PATCH 0375/1239] Config: remove redundant definitions

---
 include/configs/thunderx_88xx.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/configs/thunderx_88xx.h b/include/configs/thunderx_88xx.h
index 8974fba8a1..eaddbf22a5 100644
--- a/include/configs/thunderx_88xx.h
+++ b/include/configs/thunderx_88xx.h
@@ -12,8 +12,6 @@
 
 /*#define CONFIG_ARMV8_SWITCH_TO_EL1*/
 
-#define CONFIG_SYS_GENERIC_BOARD
-
 #define CONFIG_SYS_64BIT
 
 #define CONFIG_SYS_NO_FLASH
@@ -173,6 +171,7 @@
 #define CONFIG_CMD_SATA
 
 #define CONFIG_CMD_ATF
+#define CONFIG_CMD_PCI
 
 /* AHCI support Definitions */
 #ifdef CONFIG_CMD_SATA
-- 
2.29.0

