From eeee4fba8a891747d3f6908bb147f92cfd72f37a Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 5 May 2016 20:44:38 -0700
Subject: [PATCH 0409/1239] Fix compilation errors.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/cavium/thunder_bgx.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index e455e4510f..bcb15a6576 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -226,7 +226,7 @@ static int get_qlm_for_bgx(int node, int bgx_id, int index)
 
 	cfg1 = readq(CSR_PA(node, GSERX_CFG(qlm))) & GSERX_CFG_BGX;
 	cfg2 = readq(CSR_PA(node, GSERX_CFG(qlm+1))) & GSERX_CFG_BGX;
-	debug("get_qlm_for_bgx:qlm%d: cfg1 = %d, cfg2 = %d\n", qlm, cfg1, cfg2);
+	debug("get_qlm_for_bgx:qlm%d: cfg1 = %lld, cfg2 = %lld\n", qlm, cfg1, cfg2);
 
 	/* Check if both DLMs are configured as BGX# */
 	if (cfg2) {
@@ -298,7 +298,7 @@ static int bgx_lmac_xaui_init(struct bgx *bgx, int lmacid, int lmac_type)
 	u64 cfg;
 	struct lmac *lmac;
 
-	lmac = &bgx[lmacid];
+	lmac = &bgx->lmac[lmacid];
 
 	/* Reset SPU */
 	bgx_reg_modify(bgx, lmacid, BGX_SPUX_CONTROL1, SPU_CTL_RESET);
@@ -605,7 +605,6 @@ static void bgx_init_hw(struct bgx *bgx)
 {
 	struct lmac *lmac;
 	int i, lmacid, lmac_count = 0;
-	uint64_t rx_lmacs;
 
 	for (lmacid = 0; lmacid < MAX_LMAC_PER_BGX; lmacid++) {
 		lmac = &bgx->lmac[lmacid];
-- 
2.29.0

