From d59b3440741a64aa0b52206529b2f148256e496a Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Tue, 7 Jun 2016 14:29:09 -0700
Subject: [PATCH 0498/1239] net: thunderx: Remove local printfs

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/cavium/thunder_xcv.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/cavium/thunder_xcv.c b/drivers/net/cavium/thunder_xcv.c
index 628de2f45a..548851d074 100644
--- a/drivers/net/cavium/thunder_xcv.c
+++ b/drivers/net/cavium/thunder_xcv.c
@@ -57,7 +57,6 @@ int xcv_setup_link(bool link_up, int link_speed)
 	credits = link_up && !reset.s.enable;
 	xcv_ctl.u = readq(CSR_PA(0, XCVX_CTL(0)));
 
-printf("link_up = %d, speed = %d, xcv_ctl.speed = %d\n", link_up, link_speed, speed, xcv_ctl.s.speed);
 	if (link_up && (!reset.s.enable || (xcv_ctl.s.speed != speed))) {
 		/* Enable the XCV block */
 		reset.u = readq(CSR_PA(0, XCVX_RESET(0)));
@@ -112,7 +111,6 @@ printf("link_up = %d, speed = %d, xcv_ctl.speed = %d\n", link_up, link_speed, sp
 		reset.s.tx_dat_rst_n = 1; 
 		reset.s.rx_dat_rst_n = 1; 
 		writeq(reset.u, CSR_PA(0, XCVX_RESET(0)));
-printf("done with rx/tx data rst\n");
 	}
 
 	/* Enable the packet flow
@@ -152,7 +150,6 @@ int thunderx_xcv_probe(struct udevice *dev)
 	}
 
 	xcv->reg_base = dm_pci_map_bar(dev, 0, &size, PCI_REGION_MEM);
-printf("xcv_reg_base = 0x%llx\n", xcv->reg_base);
 
 	return 0;
 }
-- 
2.29.0

