From 1bd6707d23616ab3bf5d29e883432929ab2a0c17 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Fri, 10 Feb 2017 14:34:29 -0800
Subject: [PATCH 0611/1239] Configure only one lmac for aqr107 interface.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/cavium/thunder_bgx.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 92d41bb36a..8759cb37d6 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -1035,6 +1035,11 @@ static void bgx_init_hw(struct bgx *bgx)
 				    && (bgx->bgx_id == 0)
 				    && (lmacid == 2))
 					continue;
+				else if (board
+					 && (strncasecmp(board, "nas", 3) == 0)
+					 && (bgx->bgx_id == 1)
+					 && (lmacid == 1))
+					continue;
 				lmac->lane_to_sds = lmacid;
 			}
 			lmac->lmac_type = 3;
@@ -1220,7 +1225,13 @@ static void bgx_get_qlm_mode(struct bgx *bgx)
 				    && (lmacid == 2)) {
 					lmac->qlm_mode = QLM_MODE_XFI;
 					continue;
-				}
+				} else if (board
+					   && (strncasecmp(board, "nas", 3) == 0)
+					   && (bgx->bgx_id == 1)
+					   && (lmacid == 1)) {
+					lmac->qlm_mode = QLM_MODE_XFI;
+					continue;
+                                }
 				lmac->qlm_mode = QLM_MODE_XFI;
 				printf("BGX%d QLM%d LMAC%d mode: XFI\n",
 					bgx->bgx_id, lmac->qlm, lmacid);
-- 
2.29.0

