From 3552d64a8ef7452e6713711db7cfe18772137d55 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Tue, 14 Nov 2017 17:24:41 -0800
Subject: [PATCH 0741/1239] ahci: limit max scsi id based on supported ports in
 controller

Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 drivers/ata/ahci.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/ata/ahci.c b/drivers/ata/ahci.c
index 1e7c50b19b..24f3a65357 100644
--- a/drivers/ata/ahci.c
+++ b/drivers/ata/ahci.c
@@ -1177,7 +1177,6 @@ int ahci_probe_scsi(struct udevice *ahci_dev, ulong base)
 	if (ret)
 		return ret;
 
-	uc_plat->max_lun = 1;
 	uc_plat->max_id = uc_priv->n_ports;
 
 	ret = ahci_start_ports(uc_priv);
-- 
2.29.0

